### DISCUSSION TITLE
Text and data can only be passed by String, ArrayBuffer or TypedArray

### DISCUSSION THREAD

**Discussion by vimalmistry** [uNetworking/uWebSockets.js#544] [Q&A] [answered]
Here is code for publish by channel.

```js
      try {
            this.app.publish(channel, jsonEncode([event, data]))
        } catch (err) {
            console.log("SEND ERROR", { err, send_data: jsonEncode([event, data]) }, typeof jsonEncode([event, data]))
        }
```

The issue is Its sending data everytime. But Some of message is not working. and giving me error 
`Text and data can only be passed by String, ArrayBuffer or TypedArray`

Working..
```
["user_akn",{"usr":{"id":148,"name":"Andris","email":"acrevy43@t-online.de","gender":"Genderqueer","lat":7.3423696,"lng":-76.5719414},"id":1}]
```

Not Working.
![Screenshot 2021-06-05 at 9 33 21 PM](https://user-images.githubusercontent.com/3993163/120897815-a8162a00-c645-11eb-8ec4-09f23a26e801.png)

As you can see third value it's shows typeof send data is string. Strange but What's the problem? Can somebody guide me plz. I am stuck here for 3 hours.

**Comment by ghost** [comment, marked as answer]
If a and b take string and you have confirmed b is string then.............. maybe confirm a is string? It is impossible to make out anything from this snippet

**Reply by vimalmistry** [reply]
Wow.
I just changed, Now it's working. It's require channel name to be string.
```js
            this.app.publish(channel.toString(), msg)
```

**Reply by ghost** [reply]
Right so 3 hours wasted to JavaScript lack of types.... so _productive_

**Reply by e3dio** [reply]
3 hours wasted due to gross incompetence not lack of types, prob spent most his time trying to figure out where it threw https://github.com/uNetworking/uWebSockets.js/pull/542

**Reply by ghost** [reply]
That's one way to put it 😄

**Reply by joshxyzhimself** [reply]
> Right so 3 hours wasted to JavaScript lack of types.... so productive

This is so true lol, always gets me.

@vimalmistry you might wanna try TypeScript's `// @ts-check` directive along with the use of `*.d.ts` files so you can easily catch these type-errors even if your code is in JavaScript.

**Reply by vimalmistry** [reply]
Yes. It is look like foolish. But It's happen in js world. I seen peoples tinkering their head for hours because they forgot the async before function call.
